core/arch/arm/plat-imx/sm_platform_handler.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <io.h>
+#include <kernel/cache_helpers.h>
+#include <kernel/tz_ssvce_pl310.h>
+#include <mm/core_memprot.h>
+#include <platform_config.h>
+#include <sm/sm.h>
+#include <sm/optee_smc.h>
+
+#include "smc_sip.h"
+
+#ifdef CFG_PL310
+static void handle_sip_pl310(struct thread_smc_args *smc_args)
+{
+	smc_args->a0 = OPTEE_SMC_RETURN_OK;
+	DMSG("SIP L2C310 val=%x addr=%x\n", smc_args->a1,
+	     smc_args->a2);
+	/*
+	 * Disable L2 is only needed in low power case,
+	 * the psci asm code will take care of disable.
+	 * So we just flush data here.
+	 */
+	if (smc_args->a1 == 0 && smc_args->a2 == PL310_CTRL) {
+		dcache_op_all(DCACHE_OP_CLEAN);
+		arm_cl2_cleanbyway(pl310_base());
+		return;
+	}
+	switch (smc_args->a2) {
+	case PL310_CTRL:
+	case PL310_PREFETCH_CTRL:
+	case PL310_AUX_CTRL:
+		write32(smc_args->a1, pl310_base() + smc_args->a2);
```

Comment:
I think you need to filter out some bit fields for `PL310_PREFETCH_CTRL` and `PL310_AUX_CTRL` to insure consistency of the effective configuration and the configuration expected by the TEE.

Also for the `PL310_CTRL` otherwise one can disable the L2 by setting  for example `smc_args->a1 = 2`.